app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,102 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.order(id: :asc)
+  end
+
+  def show
+    # = is assigning          # accessing
+    # @tasks = Task.where(id: params[:id]).order(id: :asc)
+
+    # where returns an array that can have more then one element
+    # find returns one instance of the class
+    # will throw ActiveRecord::RecordNotFound: Couldn't find Task with 'id'=500
+    @tasks = Task.find(params[:id])
+
+    # take html.erb page and give it to the browser
+    # render :show
+  end
+
+  # this creates an empty shell all the viewer sees is the link name
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new(task_create_params[:task]) #instantiate a new task
+    if @task.save  #save returns true if the database insert succeeds
+      #begins new request,
+      redirect_to root_path # go to the index so we can see the album in the list
+    else  # if save failed
+      flash[:error] = @task.errors.full_messages
```

Comment:
Awesome! You're checking to see if the task has the proper attributes and flashing an error if not-- that's a great user experience!